### PR TITLE
feat(finetuner): add adjustable warmup

### PR DESCRIPTION
**Pull request by harubaru**
This PR adds a new argument to #128 which allows clients to adjust the ratio of warm up steps to training steps which is fairly important since the GPT finetuner uses an adaptive optimizer (Adam), plus this also addresses @Eta0's feedback on a duplicate workflow argument.

### CODE REVIEW

File: finetuner-workflow/finetune-workflow.yaml [hunk 1]
```diff
@@ -275,6 +275,7 @@ spec:
               --train-ratio=1.0
               --eot='{{workflow.parameters.eot_token}}'
               --pad='{{workflow.parameters.pad_token}}'
+              --warmup_ratio={{workflow.parameters.warmup_ratio}}
```
Comment (Eta0) [inline finetuner-workflow/finetune-workflow.yaml:278]:
```suggestion
              --warmup-ratio={{workflow.parameters.warmup_ratio}}
```

File: finetuner-workflow/finetuner/finetuner.py [hunk 2]
```diff
@@ -988,7 +999,7 @@ def evaluate(
     gradient_checkpointing=True,
 
     # Learning rate scheduler arguments
-    warmup_steps=8,
+    warmup_steps=warmup_steps,
```
Comment (Eta0) [inline finetuner-workflow/finetuner/finetuner.py:1002]: `TrainingArguments` supports specifying the `warmup_ratio` directly as a float:
```suggestion
    warmup_ratio=float(args.warmup_ratio),
```

File: finetuner-workflow/finetuner/finetuner.py [hunk 3]
```diff
@@ -972,6 +979,10 @@ def evaluate(
         )
     )
 
+# Calculate how many warmup steps we need to do using the warmup ratio.
+training_steps = int((args.epochs * len(train_dataset)) / (args.gradients * bs * world_size))
+warmup_steps = int(args.warmup_ratio * training_steps)
+
```
Comment (Eta0) [inline finetuner-workflow/finetuner/finetuner.py:985]: This calculation can be skipped by specifying `warmup_ratio` directly in the `TrainingArguments` constructor (see other comment)
```suggestion
```

**Review by Eta0 — CHANGES_REQUESTED** [review]
Left some comments on how it can be simplified a bit, but besides that, it looks like a good change!

File: finetuner-workflow/finetuner/finetuner.py (hunk 3 above)
Comment (harubaru) [reply finetuner-workflow/finetuner/finetuner.py:985]: Oh wow didn't know that

**Review by Eta0 — APPROVED** [review]

**Review by wbrown — APPROVED** [review]
Good. :+1:
